### PR TITLE
Fix firmware updating/reinstallation

### PR DESCRIPTION
**Pull request by elad335**
We forgot to close Qt's file descriptors of PS3 firmware fonts we use when they are present. So attempting to open another file descriptor for writing to the PS3 fonts' file in order to install the file failed with access violation error. Of course reinitialize the Qt fonts after installation has finished.

### CODE REVIEW

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -787,6 +788,24 @@ void main_window::HandlePupInstallation(QString file_path)
...
+			: mw(_mw)
+		{
+			// Remove possibly PS3 fonts from database
+			QFontDatabase::removeAllApplicationFonts();
+		}
+
+		~fonts_manage()
+		{
+			// Update with newly installed PS3 fonts
+			Q_EMIT mw->RequestGlobalStylesheetChange();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:805]: just call this before creating the firmware cache. no need for some weird struct.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 2]
```diff
@@ -787,6 +788,24 @@ void main_window::HandlePupInstallation(QString file_path)
...
 
+	struct fonts_manage
+	{
+		main_window* mw;
+
+		fonts_manage(main_window* _mw)
+			: mw(_mw)
+		{
+			// Remove possibly PS3 fonts from database
+			QFontDatabase::removeAllApplicationFonts();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:799]: and call this in gui_application::OnChangeStyleSheetRequest (just move the existing call to the top of the function)

File: rpcs3/rpcs3qt/main_window.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp:805]: It's for proper error handling, noone said installation always succeeds.

Comment (Megamouse) [reply rpcs3/rpcs3qt/main_window.cpp:805]: it doesn't make any sense. there are at least 5 returns in this function before we even start installing the firmware

Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp]: Nvm, fixed.

File: rpcs3/rpcs3qt/main_window.cpp (hunk 2 above)
Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp]: Like this? or after the CLI check?

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 3]
```diff
@@ -451,6 +451,21 @@ void gui_application::StopPlaytime()
 */
 void gui_application::OnChangeStyleSheetRequest()
 {
+	const QString config_dir = qstr(fs::get_config_dir());
+
+	// Remove old fonts
+	QFontDatabase::removeAllApplicationFonts();
+
+	// Add PS3 fonts
+	QDirIterator ps3_font_it(qstr(g_cfg.vfs.get_dev_flash() + "data/font/"), QStringList() << "*.ttf", QDir::Files, QDirIterator::Subdirectories);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_application.cpp:460]: why did you move it all up ?
none of these fonts are used if we don't actually load a stylesheet

Comment (elad335) [reply rpcs3/rpcs3qt/gui_application.cpp]: Fixed now?

Comment (Megamouse) [reply rpcs3/rpcs3qt/gui_application.cpp]: what are you even doing???

**Review by Megamouse — APPROVED** [review]
